pallets/registrar/src/lib.rs

```diff
@@ -139,7 +139,7 @@ pub mod pallet {
         type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
         /// Origin that is allowed to call register and deregister
-        type RegistrarOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+        type RegistrarOrigin: EnsureOriginWithArg<Self::RuntimeOrigin, ParaId>;
```

Comment:
The comment is misleading, register uses signed origin (not configurable), and this origin is used for deregister and other management calls such as pause and set_parathread_params

Reply:
Updated, is it a bit more clear?